includes/server/llms-rest-server-functions.php

```diff
@@ -66,6 +66,46 @@ function llms_rest_not_found_error( $message = '' ) {
 	return new WP_Error( 'llms_rest_not_found', $message, array( 'status' => 404 ) );
 }
 
+/**
+ * Wrapper for `llms_page_restricted()` which can be used during REST requests
+ *
+ * Since the `$post` and `$wp_query` globals aren't set during REST requests the
+ * `llms_page_restricted()` method doesn't work as expected since it relies on WP core
+ * template tags (like `is_singular()`) which do not work during REST requests.
+ *
+ * This is probably a hacky workaround that will be removed in the future. A better
+ * solution would be to refactor `llms_page_restricted()` to not rely on conditional
+ * tags.
+ *
+ * @since [version]
+ *
+ * @param int $post_id WP_Post ID.
+ * @param int $user_id WP_User ID.
+ * @return array Hash of restriction data from `llms_page_restricted()`.
+ */
+function llms_rest_page_restricted( $post_id, $user_id ) {
+
+	global $post, $wp_query;
+
+	// Store preexisting global values.
+	$temp_post        = $post;
+	$temp_is_singular = $wp_query->is_singular;
+
+	// Override them.
+	$post                  = get_post( $post_id );
+	$wp_query->is_singular = true;
```

Comment:
I was following this approach on my branch https://github.com/eri-trabiccolo/lifterlms-rest/tree/restrictions
https://github.com/eri-trabiccolo/lifterlms-rest/commit/debe8b5485400ccb10d70cbd987e0dfcba505c7f#diff-5ce7aeb33633a2220e61908290ff5aedR806-R815

I needed to set the query's `queried_object` and the `queried_object_id` as well, I cannot remember exactly why at the moment. But just in case you'd find some inconsistencies during your tests, take this into account.

Reply:
I thought I saw you start doing something like this somewhere but I couldn't find it!

Reply:
And the reason I added this as a function was because I was pretty sure you were doing something like this and figured we'd need it elsewhere at somepoint

Reply:
I think you did it right.
I did it inside the "loop" in the post controller because I needed it in the posts loop but I will adapt my code using the new function. Looks more clean.

Reply:
@eri-trabiccolo I'm working on reworking `llms_page_restricted()` to be useable outside a loop. This function is some very stringy code and needs a refactor. I'm taking it on instead of writing a hack around it.